examples/DotNetCore/EventHub/Controllers/HomeController.cs

```diff
@@ -0,0 +1,169 @@
+
+            //
+            // Check if the required configurations are available.
+            if (string.IsNullOrEmpty(eventHubConnectionString))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnectionString));
+            }
+
+            if (string .IsNullOrEmpty(eventHubName))
+            {
+                throw new ArgumentNullException(nameof(eventHubName));
+            }
+
+            if (string.IsNullOrEmpty(blobStorageConnectionString))
+            {
+                throw new ArgumentNullException(nameof(blobStorageConnectionString));
+            }
+            
+            if (string.IsNullOrEmpty(blobContainerName))
+            {
+                throw new ArgumentNullException(nameof(blobContainerName));
+            }
+
+            //
+            // If no consumer group is provided, use default one.
+            if (string.IsNullOrEmpty(consumerGroup))
+            {
+                consumerGroup = EventHubConsumerClient.DefaultConsumerGroupName;
+            }
+
+            //
+            // Create blob storage client.
+            var storageClient = new BlobContainerClient(connectionString: blobStorageConnectionString,
+                                                        blobContainerName: blobContainerName);
+            
+            _processorClient = new EventProcessorClient(checkpointStore: storageClient,
+                                                        consumerGroup: consumerGroup,
+                                                        connectionString: eventHubConnectionString,
+                                                        eventHubName: eventHubName);
+        }
+
+        private string ReadFromConfigStore(string key, string label)
+        {
+            if (string.IsNullOrWhiteSpace(label))
+            {
+                label = LabelFilter.Null;
+            }
+
+            var builder = new ConfigurationBuilder();
+            builder.AddAzureAppConfiguration(options =>
+            {
+                options.Connect(_eventHubConnection.AppConfigConnectionString)
+                        .Select(key, label);
+            });
+
+            IConfiguration configuration = builder.Build();
+            return configuration[key];
+        }
+
+        private Task ProcessEventHandler(ProcessEventArgs eventArgs)
+        {
+            string rawEvent = Encoding.UTF8.GetString(eventArgs.Data.Body.ToArray());
+            IEnumerable<EventContract> events = JsonConvert.DeserializeObject<List<EventContract>>(rawEvent);
+
+            foreach (var e in events)
+            {
+                var key = e.Data.Key;
+                var label = e.Data.Label;
+
+                if (string.Equals(key, "Messages", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    var value = ReadFromConfigStore(key, label);
+                    Settings.Messages = value;
```

Comment:
>Settings.Messages = value [](start = 20, length = 25)

Updating object model directly due to update in configuration sounds like a misstep to me. I would expect that the push model still be integrated with IConfiguration. It just exposes a new way to notify the application that it needs to refresh it's IConfiguration.

Reply:
Changed to call the refresher instead.